src/app/content/highlights/components/CardWrapper.tsx

```diff
@@ -68,23 +70,24 @@ const Wrapper = ({highlights, className, container, highlighter}: WrapperProps)
   }, [cardsHeights]);
 
   const getOffsetsForHighlight = React.useCallback((highlight: Highlight) => {
-    if (offsets.has(highlight.id)) {
-      return assertDefined(offsets.get(highlight.id), 'this has to be defined');
-    } else {
-      const newOffsets = assertDefined(
-        getHighlightOffset(container, highlight),
-        `Couldn't get offsets for highlight with an id: ${highlight.id}`
-      );
-      setOffsets((state) => new Map(state).set(highlight.id, newOffsets));
-      return newOffsets;
-    }
-  }, [container, offsets]);
+    const newOffsets = assertDefined(
+      getHighlightOffset(container, highlight),
+      `Couldn't get offsets for highlight with an id: ${highlight.id}`
+    );
+    setOffsets((state) => new Map(state).set(highlight.id, newOffsets));
+    return newOffsets;
+  }, [container]);
 
-  React.useEffect(() => {
+  const setNewCardsPositions = React.useCallback(() => {
     const positions = updateCardsPositions(focusedHighlight, highlights, cardsHeights, getOffsetsForHighlight);
     setCardsPositions(positions);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [highlights, cardsHeights, focusedHighlight]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [cardsHeights, focusedHighlight, highlights]);
```

Comment:
I think it is okay to add `getOffsetsForHighlight` to the deps array since now it will change only once per container